apps/aechannel/src/aesc_fsm_min_depth_watcher.erl

```diff
@@ -13,46 +14,202 @@
+                  , {height       , block_hash()} => non_neg_integer()
+                  , {header       , block_hash()} => aec_headers:header()
+                  , {in_main_chain, block_hash()} => boolean()
+                  , {channel      , block_hash()} => aesc_channels:channel()
+                  , {tx_hashes    , block_hash()} => [tx_hash()] }.
+
+watch_for_channel_close(Pid, MinDepth, Mod) when is_pid(Pid) ->
+    gen_server:call(Pid, close_req(MinDepth, Mod));
 watch_for_channel_close(ChanId, MinDepth, Mod) ->
     gen_server:start_link(?MODULE, #{parent  => self(),
                                      chan_id => ChanId,
-                                     requests =>
-                                         [#{mode         => close,
-                                            min_depth    => MinDepth,
-                                            type         => close,
-                                            parent       => self(),
-                                            callback_mod => Mod}]},
+                                     requests => [close_req(MinDepth, Mod)]},
                           ?GEN_SERVER_OPTS).
 
+watch_for_unlock(Pid, Mod) when is_pid(Pid) ->
+    gen_server:call(Pid, unlock_req(Mod));
+watch_for_unlock(ChanId, Mod) ->
+    gen_server:start_link(?MODULE, #{parent => self(),
+                                     chan_id => ChanId,
+                                     requests => [unlock_req(Mod)]},
+                          ?GEN_SERVER_OPTS).
+
+watch_for_min_depth(Pid, TxHash, MinDepth, Mod, Info) when is_pid(Pid) ->
+    gen_server:call(Pid, min_depth_req(TxHash, MinDepth, Mod, Info)).
+
+close_req(MinDepth, Mod) ->
+    #{ mode         => close
+     , min_depth    => MinDepth
+     , info         => #{ type  => close
+                        , parent => self()
+                        , callback_mod => Mod } }.
+
+unlock_req(Mod) ->
+    #{mode => unlock,
+      info => #{ parent       => self()
+               , type         => closing
+               , callback_mod => Mod }}.
+
+min_depth_req(TxHash, MinDepth, Mod, Type) ->
+    #{ mode      => tx_hash
+     , min_depth => MinDepth
+     , tx_hash   => TxHash
+     , info      => #{ type         => Type
+                     , parent       => self()
+                     , callback_mod => Mod }}.
+
+get_txs_since({all_after_tx, _Hash} = StopCond, ChId) ->
+    get_txs_since_(StopCond, ChId);
+get_txs_since({any_after_block, _Hash} = StopCond, ChId) ->
+    get_txs_since_(StopCond, ChId).
+
+get_txs_since_(StopCond, ChId) ->
+    get_txs_since(StopCond, aec_chain:top_block_hash(), ChId, #{}).
+
+
 start_link(Type, TxHash, ChanId, MinDepth, Mod) ->
+    I = #{ parent       => self()
+         , type         => Type
+         , callback_mod => Mod },
+    Reqs = [#{ mode      => tx_hash
+             , tx_hash   => TxHash
+             , min_depth => MinDepth
+             , info      => I},
+            #{ mode  => watch
```

Comment:
I see the `init` function sends to itself `check_status` message and places these `Reqs` in `#st.requests`. Ok.